src/context.ts

```diff
@@ -44,14 +44,14 @@ export class DBOSContextImpl implements DBOSContext {
   readonly logger: DBOSLogger;      // Wrapper around the global logger for this context.
 
   constructor(readonly operationName: string, readonly span: Span, logger: Logger, parentCtx?: DBOSContextImpl) {
-    this.logger = new DBOSLogger(logger, this);
     if (parentCtx) {
       this.request = parentCtx.request;
       this.authenticatedUser = parentCtx.authenticatedUser;
       this.authenticatedRoles = parentCtx.authenticatedRoles;
       this.assumedRole = parentCtx.assumedRole;
       this.workflowUUID = parentCtx.workflowUUID;
     }
+    this.logger = new DBOSLogger(logger, this);
```

Comment:
This fix a bug where the metadata would be missing if provided from the `parentCtx`